pylint/checkers/similar.py

```diff
@@ -137,10 +138,16 @@ def stripped_lines(lines, ignore_comments, ignore_docstrings, ignore_imports):
     """return lines with leading/trailing whitespace and any ignored code
     features removed
     """
+    if ignore_imports:
+        tree = ast.fix_missing_locations(ast.parse(''.join(lines)))
+        line_is_import = {
+            node.lineno: isinstance(node, (ast.Import, ast.ImportFrom))
+            for node in tree.body}
```

Comment:
This results in the final expression of the line overwriting any previous ones. For example:

```python
from foo import bar; print (bar)
```

This results in a `line_is_import` of:
```
{1: False}
```

It's horrible code, however it is legal python. Something like this may work:

```python
expression_is_import = [
    (node.lineno, isinstance(node, (ast.Import, ast.ImportFrom)))
    for node in tree.body]
line_is_import = {
    lineno: any(is_import for _, is_import in lineno_is_import)
    for lineno, lineno_is_import in groupby(expression_is_import, key=lambda expression: expression[0])}
```

Reply:
I see what you mean, but I think `any` should be `all`: --ignore-imports should only ignore imports; the presence of any code should keep the line under consideration.